### DISCUSSION TITLE
Interfacing pyGIMLi with MATLAB

### DISCUSSION THREAD

**Discussion by ruboerner** [gimli-org/pyGIMLi#723] [Q&A]
Hello all,

I am trying to incorporate a simple pyGIMLi-based inversion into a MATLAB based GUI that controls our in-house multi-electrode instrument.

So far, I can import pygimli and all relevant modules.

An MWE looks like this:

```
pyenv( ...
    Version="/Users/rub/miniconda3/envs/pg/bin/python", ...
    ExecutionMode="OutOfProcess")

pg = py.importlib.import_module('pygimli');
ert = py.importlib.import_module('pygimli.physics.ert');
mgr = ert.ERTManager('mactest.mea');
model = mgr.invert(lam=20);

mgr.showResult() % MATLAB crashes
```

The last line in the MATLAB script causes MATLAB to crash.

Has anybody made some progress in interfacing pyGIMLi and MATLAB?

Thanks in advance!

**Comment by halbmy** [comment]
I just tested under Windows
```
pyenv(Version="c:\Program Files\software\Anaconda\envs\pg15\python.exe", ExecutionMode="OutOfProcess")
ert = py.importlib.import_module('pygimli.physics.ert');
plt = py.importlib.import_module('matplotlib.pyplot');
data = ert.load('gallery.dat');
data.estimateError()
mgr = ert.Manager(data);
mgr.invert(verbose=true);
mgr.showResult();
plt.show();
```
and it shows the well-known gallery result.

We found that the problem already occurs (on the used Mac) when importing matplotlib and showing a figure, so it is independent of pyGIMLi. Nevertheless it is a good idea that such a thing exists. Thank you for reporting!
